### PR TITLE
Add build deb and rpm packages.

### PR DESCRIPTION
**Pull request by anton-bystrov**
[PMM-5598](https://jira.percona.com/browse/PMM-5598)

Add building deb and rpm packages.

---

- [ ] Tests passed.
- [x] Fix conflicts with target branch.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: .scripts/systemd/mongodb_exporter.service [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+After=network.target
+
+[Service]
+Type=simple
+
+User=pmm
+Group=pmm
+
+ExecStart=/usr/bin/mongodb_exporter \
+  --mongodb.uri=mongodb://user:pass@127.0.0.1:27017/admin?ssl=true \
```
Comment (denisok) [inline .scripts/systemd/mongodb_exporter.service:13]: interesting... this probably should come from the env variables or conf file.

File: .goreleaser.yml [hunk 2]
```diff
@@ -40,3 +40,23 @@ dockers:
...
+    maintainer: Percona
+    vendor: Percona
+    license: Apache 2.0
+    formats:
+    - deb
+    - rpm
+    bindir: /usr/bin
+    replacements:
+      amd64: 64-bit
+      darwin: macOS
```
Comment (denisok) [inline .goreleaser.yml:56]: is this needed for deb and rpm?

File: .scripts/postinst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#     installation fails and the `postinst' is called with `abort-upgrade',
+#     `abort-remove' or `abort-deconfigure'.
+
+echo "Creating user and group..."
+
+GROUPNAME="pmm"
+USERNAME="pmm"
+
+getent group "$GROUPNAME" >/dev/null || groupadd -r "$GROUPNAME"
+getent passwd "$USERNAME" >/dev/null || useradd -r -g "$GROUPNAME" -M -s /sbin/nologin -c "pmm service user" "$USERNAME"
```
Comment (nikita-b) [inline .scripts/postinst:29]: It looks overcomplicated. You can use single command:
``` 
   adduser --quiet --no-create-home  --group -c"PMM User" pmm
```

**Comment by nikita-b** [comment]
What's about the defaults file?

File: .scripts/postinst [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+#          <conflicting-package> <version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+#
+# quoting from the policy:
+#     Any necessary prompting should almost always be confined to the
+#     post-installation script, and should be protected with a conditional
+#     so that unnecessary prompting doesn't happen if a package's
+#     installation fails and the `postinst' is called with `abort-upgrade',
+#     `abort-remove' or `abort-deconfigure'.
```
Comment (nikita-b) [inline .scripts/postinst:21]: Do we need this comment?

File: .scripts/postinst [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+echo "Creating user and group..."
+
+GROUPNAME="pmm"
+USERNAME="pmm"
+
+getent group "$GROUPNAME" >/dev/null || groupadd -r "$GROUPNAME"
+getent passwd "$USERNAME" >/dev/null || useradd -r -g "$GROUPNAME" -M -s /sbin/nologin -c "pmm service user" "$USERNAME"
+
+systemctl daemon-reload > dev/null
```
[the remark this answers is not in the document]
Comment (nikita-b) [reply .scripts/postinst:31]: Debian has a lot of helper scripts. For example:
http://manpages.ubuntu.com/manpages/bionic/man1/deb-systemd-invoke.1p.html

The trick is that we need RPM and DEB packages together...

FPM can create systemd service itself but it seems that goreleaser NFPM can't...

**Comment by nikita-b** [comment]
A couple of points:
1. We need to register a systemd service.
2. We need to support upgrading to a new version (even for systemd service)
3.  We need to give the user the ability to overwrite the default settings (default file)

**Review by nikita-b — CHANGES_REQUESTED** [review]

**Comment by denisok** [comment]
> A couple of points:
> 
>     1. We need to register a systemd service.
> 
>     2. We need to support upgrading to a new version (even for systemd service)
> 
>     3. We need to give the user the ability to overwrite the default settings (default file)

maybe we are over complicating it.  Do we really need service file on thi stage? mongo exporter itself doesn't support environment or passing config file or parameter as a file (see #380).

I would probably remove service and get service later when mongodb_exporter is better parametrized, like taking user, url, pass from env.

File: .goreleaser.yml (hunk 2 above)
Comment (anton-bystrov) [reply .goreleaser.yml:56]: I think this naming is more beautiful...no more.

**Review by denisok — APPROVED** [review]

**Review by nikita-b — APPROVED** [review]

**Pull request merged**